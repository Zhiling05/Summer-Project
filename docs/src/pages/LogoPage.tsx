<<<<<<< HEAD
function LogoPage() {
  return <h1>Logo Page</h1>;
}

export default LogoPage;
=======
import { useNavigate } from 'react-router-dom';
import '../styles/question.css';        // 引入 NHS 头尾、按钮 等样式
import NHSLogo from '../assets/NHS_LOGO.jpg';
import DIPPLogo from '../assets/DIPP_Study_logo.png';

export default function LogoPage() {
  const navigate = useNavigate();

  return (
    <>
      {/* ---------- 顶部栏 ---------- */}
      <header className="nhs-header">
        <div className="nhs-header__inner">
          <img className="logo nhs-logo" src={NHSLogo} alt="NHS logo" />
          <img className="logo dipp-logo" src={DIPPLogo} alt="DIPP Study logo" />
          <span className="nhs-header__service">DIPP Study</span>
        </div>
      </header>

      {/* ========== 主体 ========== */}
      <div className="nhsuk-width-container">
        <main id="maincontent">
          <section className="question-box">
            <h1 className="nhsuk-heading-l">Welcome to the DIPP Study</h1>
            <p className="nhsuk-body">
              This tool will guide you through the DIPP referral pathway assessment,
              helping you identify red-flag symptoms, produce consensus
              recommendations, and ultimately improve patient care.
            </p>
            <ul className="nhsuk-list nhsuk-list--bullet">
              <li>Identify how optometrists manage patients with potential papilloedema</li>
              <li>Co-produce and trial consensus guidelines with PPI input</li>
              <li>Evaluate the impact on NHS resources and patient experience</li>
            </ul>

            <button
              className="continue-button"
              onClick={() => navigate('/optometrist/assess/start-page')}
            >
              Start now
            </button>
          </section>
        </main>
      </div>

      {/* ========== 页脚 ========== */}
      <footer className="nhs-footer">
        <div className="footer-inner">
          <p>
            Other ways to contact DIPP if you have a hearing problem or need help in other languages&nbsp;
            <a href="#/" target="_blank" rel="noopener noreferrer">(opens in a new tab)</a>.
          </p>
          <hr />
          <p>The following links open in a new tab:</p>
          <ul className="footer-links">
            <li><a href="#/" target="_blank" rel="noopener noreferrer">Privacy statement</a></li>
            <li><a href="#/" target="_blank" rel="noopener noreferrer">Terms and conditions</a></li>
            <li><a href="#/" target="_blank" rel="noopener noreferrer">Accessibility statement</a></li>
          </ul>
        </div>
      </footer>
    </>
  );
}
>>>>>>> remotes/origin/Junjie_develop
