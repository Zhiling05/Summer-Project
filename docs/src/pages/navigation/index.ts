// 这个文件的功能：把导航栏里的组件放在一个index.ts中，方便一次性导入。
export { default as SettingsPage } from './SettingsPage';
export { default as ContactUsPage } from './ContactUsPage';
export { default as AboutUsPage } from './AboutUsPage';


// 当你想要导入时只用写以下代码：
// import { SettingsPage, ContactUsPage, AboutUsPage } from './pages/navigation';