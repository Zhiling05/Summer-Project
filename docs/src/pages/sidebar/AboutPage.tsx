// src/pages/sidebar/AboutPage.tsx
import Header from '../../components/Header';
import BackButton from '../../components/BackButton';


export default function AboutPage() {
  return (
    <div className="about-page">
      <Header title="About Us" />
            <BackButton />
      <h1 className="about-title">About Us Page</h1>
      
      <section className="about-section">
        <h2 className="about-heading">DIPP Research Team</h2>
        <p className="about-description">
          The DIPP is led by a team of passionate researchers dedicated to improving healthcare.
        </p>
        
        <h3 className="team-title">Team members:</h3>
        <ul className="team-list">
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
          <li>XXX - XXXXXXX</li>
        </ul>
      </section>
    </div>
  );
}
