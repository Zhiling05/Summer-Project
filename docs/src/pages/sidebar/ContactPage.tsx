// src/pages/sidebar/ContactPage.tsx
export default function ContactPage() {
  return <h1>Contact us page</h1>;
}
