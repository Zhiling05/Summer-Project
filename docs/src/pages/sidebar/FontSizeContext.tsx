import React, { createContext, useState, useContext } from "react";

/* Font size type definition */
type FontSizeType = "13px" | "18px" | "24px";

/* Create context for font size management */
const FontSizeContext = createContext<{
  fontSize: FontSizeType;
  setFontSize: React.Dispatch<React.SetStateAction<FontSizeType>>;
} | undefined>(undefined);

/* FontSizeProvider component for wrapping app with font size context */
export const FontSizeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [fontSize, setFontSize] = useState<FontSizeType>("18px");

  return (
    <FontSizeContext.Provider value={{ fontSize, setFontSize }}>
      {children}
    </FontSizeContext.Provider>
  );
};

/* Custom hook to access font size and setter function */
export const useFontSize = () => {
  const context = useContext(FontSizeContext);
  if (!context) {
    throw new Error('useFontSize must be used within a FontSizeProvider');
  }
  return context;
};