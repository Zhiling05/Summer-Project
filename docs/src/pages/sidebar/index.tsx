// src/pages/navigation/index.ts
export { default as SettingsPage } from './SettingsPage';
export { default as AboutUsPage }    from './AboutUsPage';
export { default as ContactPage }  from './ContactPage';
