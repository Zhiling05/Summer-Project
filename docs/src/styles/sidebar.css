:root {
  --nhs-blue: #005eb8; /* 初始蓝色，适用于浅色模式 */
  --nhs-focus-yellow: #ffeb3b;
  --sidebar-width: 280px;
  --header-height: 64px;
  --transition-speed: 0.3s;
  --page-width: 90%; /* 页面宽度为记录页面的一致宽度 */
  --content-max-width: 1200px; /* 设置内容最大宽度 */
  --side-padding: 24px; /* 页面左右边距 */
  --space-s: 8px;
  --space-m: 16px;
  --space-l: 24px;
  --core-blue: #005eb8; /* 核心蓝色 */
  --text-body: #231f20; /* 正文颜色 */
  --text-inverse: #ffffff; /* 白色文字 */
  --border-grey: #aeb7bd; /* 浅色模式下的边框颜色 */
  --card-bg: #ffffff; /* 卡片背景色（浅色模式） */
  --card-bg-dark: #273036; /* 卡片背景色（暗色模式） */
}

/* ——— Off-canvas Sidebar ——— */
.sidebar {
  display: grid;
  grid-template-rows: auto 1fr auto; /* 顶部、内容、底部 */
  gap: 16px;
  position: fixed;
  top: 0;
  left: 0;
  height: 100dvh;
  width: var(--sidebar-width);
  background: var(--nhs-blue); /* 浅色模式背景 */
  color: #fff;
  padding: 96px 24px 32px;
  box-shadow: 2px 0 8px rgba(0, 0, 0, 0.2);
  transform: translateX(-100%); /* 初始状态隐藏 */
  transition: transform var(--transition-speed) ease;
  z-index: 999;
}

.sidebar.open {
  transform: translateX(0); /* 打开时显示 */
}

/* 链接区域优化 */
.sidebar-links {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.sidebar-links li {
  margin-bottom: 24px;
}

.sidebar-links a {
  display: flex;
  align-items: center;
  gap: 12px;
  font-size: 1.15rem;
  font-weight: 600;
  color: rgba(255, 255, 255, 0.9); /* 链接文字颜色 */
  text-decoration: none;
  padding: 12px 16px;
  border-radius: 6px;
  transition: all 0.2s ease;
}

.sidebar-links a:hover,
.sidebar-links a:focus {
  background: rgba(255, 255, 255, 0.15);
  color: #fff;
  text-decoration: none;
  transform: translateX(4px);
}

/* ——— 移动端：宽度 75vw ——— */
@media (max-width: 480px) {
  :root {
    --sidebar-width: 75vw; /* 移动端侧边栏宽度为 75% 屏幕宽度 */
  }
}

/* 遮罩效果 */
.sidebar-overlay {
  position: fixed;
  inset: 0;
  background: rgba(0, 0, 0, 0.4);
  backdrop-filter: blur(2px);
  z-index: 998;
}

/* ——— 菜单按钮（汉堡按钮） ——— */
.hamburger-btn {
  position: absolute;
  top: 16px;
  left: 30px;
  width: 45px;
  height: 35px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: transparent;
  border: none;
  padding: 0;
  cursor: pointer;
  z-index: 1000; /* 确保它在顶部 */
}

.hamburger-btn .bar {
  height: 4px;
  width: 80%;
  background: #ffffff; /* 汉堡图标始终为白色 */
  border-radius: 5px;
  transition: transform var(--transition-speed) ease, opacity var(--transition-speed) ease;
}

/* 页面容器（适应侧边栏的布局） */
.contact-page-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  max-width: var(--content-max-width);
  margin: 0 auto; /* 水平居中 */
  padding: var(--space-m);
  box-sizing: border-box;
  padding-top: calc(var(--header-height) + var(--space-m));
  transition: margin-left var(--transition-speed) ease;
  min-height: 60vh;
  margin-top: -20px; /* 确保顶部无额外空白 */
}

/* 内容区域的样式 */
.contact-page-container .text-content {
  background-color: var(--card-bg); /* 使用背景色变量 */
  padding: var(--space-l);
  border-radius: 10px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  width: 100%;
  max-width: 900px; /* 最大宽度 */
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  word-wrap: break-word; /* 强制换行，防止溢出 */
  overflow-wrap: break-word;
}

/* ——— 深色模式下的样式覆盖 ——— */
@media (prefers-color-scheme: dark) {
  :root {
    --nhs-blue: #1d2227; /* 深色模式下的背景颜色 */
    --core-blue: #63b6ff; /* 深色模式下的蓝色 */
    --text-body: #e8edee; /* 深色模式下的文字颜色 */
    --text-inverse: #ffffff; /* 白色文字 */
    --border-grey: #768692; /* 深色模式下的边框颜色 */
    --card-bg: #273036; /* 深色模式下的卡片背景色 */
  }

  .sidebar {
    background: var(--nhs-blue); /* 深色背景 */
    color: var(--text-body);  /* 白色文字 */
  }

  .sidebar-links a {
    color: var(--text-inverse);/* 链接文字颜色 */
  }

  .sidebar-links a:hover,
  .sidebar-links a:focus {
    background: rgba(255, 255, 255, 0.15);
    color: var(--text-inverse); /* 浅色背景下的白色文字 */
  }

  .hamburger-btn .bar {
    background: #ffffff; /* 汉堡图标始终为白色 */
  }

  .sidebar-overlay {
    background: rgba(0, 0, 0, 0.6); /* 更暗的遮罩层 */
  }

  .contact-page-container .text-content {
    background-color: var(--card-bg); /* 使用深色模式下的背景色 */
    color: var(--text-inverse); /* 深色模式下的文字颜色 */
    border: 1px solid var(--border-grey); /* 边框颜色 */
  }
}

/* 响应式调整：宽屏设备 */
@media (max-width: 1024px) {
  .sidebar-open .contact-page-container {
    margin-left: 0;
  }
}

/* 响应式调整：中等屏幕设备 */
@media (max-width: 768px) {
  .contact-page-container .text-content {
    padding: 24px;
    border-radius: 12px;
  }
}

/* 响应式：进一步适应移动端 */
@media (max-width: 480px) {
  .contact-page-container .text-content {
    margin: 0 16px;
    padding: 16px;
    border-radius: 8px;
  }
}

/* —— Go Back 按钮 —— */
.back-button {
  background: transparent;
  border: transparent;
  color: var(--core-blue);
  font-size: 1.8rem;
  padding: 6px 14px;
  margin: var(--space-l) 0 var(--space-m);
  border-radius: var(--radius-s);
  cursor: pointer;
  transition: background 0.2s ease, color 0.2s ease;
  width: 200px ! important; /* 固定宽度，确保按钮宽度一致 */
  display: inline-flex;
  align-self: flex-start;
  
  /* 使按钮相对于父容器定位 */
  position: relative;
  top: 10% !important;   /* 距离父容器顶部 10px */
  left: 0 !important;   /* 距离父容器左边 10px */
  z-index: 10; /* 确保按钮位于其他内容之上 */
}

/* 悬浮时的样式 */
.back-button:hover,
.back-button:focus-visible {
  background: var(--core-blue);
  color: var(--text-inverse);
  outline: none;
  width: 150px;  /* 确保悬浮时宽度不变 */
}


/* 确保按钮宽度在悬浮时不会变化 */
.back-button:hover {
  width: auto;
}
