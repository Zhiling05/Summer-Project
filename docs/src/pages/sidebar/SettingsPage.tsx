import React, { useState } from "react";
import Header from '../../components/Header';    // 使用 Header 组件
import { useFontSize } from './FontSizeContext'; // 引用正确路径

export default function SettingsPage() {
  const { fontSize, setFontSize } = useFontSize(); // 获取当前字体大小和更新方法

  const handleFontSizeChange = (size: "13px" | "18px" | "24px") => {
    setFontSize(size);
  };

  return (
    <div>
      <Header title="Settings Page" showBack={true} /> {/* 使用 Header 中的返回按钮 */}
      <h1>Settings Page</h1>

      <section>
        <h2>Font Size</h2>
        <p>Choose your preferred font size:</p>
        <button onClick={() => handleFontSizeChange("13px")}>Small</button>
        <button onClick={() => handleFontSizeChange("18px")}>Medium</button>
        <button onClick={() => handleFontSizeChange("24px")}>Large</button>
      </section>
    </div>
  );
}
