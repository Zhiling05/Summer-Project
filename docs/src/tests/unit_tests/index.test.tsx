import * as NavigationPages from '../../pages/sidebar';

describe('navigation/index.ts exports', () => {
    it('should export SettingsPage component', () => {
        expect(NavigationPages.SettingsPage).toBeDefined();
        expect(typeof NavigationPages.SettingsPage).toBe('function');
    });

    it('should export AboutPage component', () => {
        expect(NavigationPages.AboutPage).toBeDefined();
        expect(typeof NavigationPages.AboutPage).toBe('function');
    });

    it('should export ContactPage component', () => {
        expect(NavigationPages.ContactPage).toBeDefined();
        expect(typeof NavigationPages.ContactPage).toBe('function');
    });

    it('should not export a non-existent page', () => {
        expect('NonExistentPage' in NavigationPages).toBe(false);
        // expect(Object.hasOwn(NavigationPages, 'NonExistentPage')).toBe(false);

    });

    it('exports remain consistent (snapshot)', () => {
        expect(Object.keys(NavigationPages)).toMatchSnapshot();
    });
});
