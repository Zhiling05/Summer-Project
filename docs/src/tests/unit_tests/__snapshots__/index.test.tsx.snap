// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation/index.ts exports exports remain consistent (snapshot) 1`] = `
[
  "SettingsPage",
  "AboutPage",
  "ContactPage",
]
`;
